File: Trabalho_Final/TF/Final/src/tf/Aluno.java
package tf;
import java.util.ArrayList;

public class Aluno{
	String nome="";
	int matricula = 0; 
	ArrayList<String> VecDisc = new ArrayList<String>();
	
	public Aluno(String nome, int matricula) {
		this.nome = nome;
		this.matricula = matricula;
	}
	
	public void Matricular(String cod){
		if(check(cod) == -1)
			VecDisc.add(cod);
	}
	
	public void Desmatricular(String cod){
		int ind = check(cod);
		if(ind != -1) {
			VecDisc.remove(ind);
		}
	}
	
	public int check(String cod){
		for(int i = 0; i< this.VecDisc.size(); i++){
			if(cod.equals(VecDisc.get(i)))
				return i;			
		}
		return -1;		
	}
	
	public String show(){
		String ss = this.nome + ", "+ this.matricula +": " + "[";
		if (this.VecDisc != null && !this.VecDisc.isEmpty()) {
			for(String i: this.VecDisc){
				ss += " " + i;
			}
		}
		ss += " ]";
		return ss;
	}
	
	public String getName() {
		return this.nome;
	}
	
	public int getMat() {
		return this.matricula;
	}
	
	public ArrayList<String> getDisc(){
		return this.VecDisc;
	}
	
}

File: Trabalho_Final/TF_V1/Aluno.py
class aluno(object):
     
    nome = None
    vecDisc = []

    def __init__(self, nome):
        self.nome = nome
        
    def Matricular(self, disc):
        if disc not in self.vecDisc:
            self.vecDisc.append(disc)
    
    def Desmatricular(self, nome):
        if nome in self.vecDisc:
            ind = self.checkInd(nome)
            del(self.vecDisc[ind])

    def checkInd(self, nome):
        self.vecDisc.index(nome)

    def show(self):
        ss = self.nome + ":"
        for i in range(len(self.vecDisc)):
            ss = ss + "\n" +"\t" + str(self.vecDisc[i])
        return ss

    def getDisc(self):
        return self.vecDisc